### PR TITLE
Filter out blacklisted collections

### PR DESCRIPTION
**Pull request by javier3448**
Filter out collections by list

### CODE REVIEW

File: src/api/namespaces/helpers/handlers/collections.ts [hunk 1]
```diff
@@ -1,22 +1,33 @@
 import {filterQueryArgs, RequestValues} from '../../utils';
 import {NeftyMarketContext} from '../index';
 import QueryBuilder from '../../../builder';
+import { ApiError } from '../../../error';
```
Comment (mvdschee) [inline src/api/namespaces/helpers/handlers/collections.ts:5]: `const allowedLists = ['whitelist', 'blacklist', 'verified', 'nsfw', 'scam'];`

File: src/api/namespaces/helpers/handlers/collections.ts [hunk 2]
```diff
@@ -1,22 +1,33 @@
...
                 FROM helpers_collection_list as v
             `);
+
     query.equal('v.assets_contract', ctx.coreArgs.atomicassets_account);
-    if (args.list) {
-        query.equal('v.list', args.list);
+    if (args.lists) {
+        const lists = args.lists.split(',');
+        for(const list of lists){
+            if(
```
Comment (mvdschee) [inline src/api/namespaces/helpers/handlers/collections.ts:22]: `if(!allowedLists.includes(list)) { ... }`

File: src/api/namespaces/helpers/handlers/collections.ts [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
+                list !== 'verified' && list !== 'nsfw' &&
+                list !== 'scam'
+            ){
+                throw new ApiError(`Invalid value ${list} for parameter lists`, 400);
+            }
+        }
+        query.equalMany('v.list', lists);
     }
     query.append(`
+                GROUP BY collection_name, list
```
Comment (blood0cean) [inline src/api/namespaces/helpers/handlers/collections.ts:34]: grouping by list will still return the collection more than once, or is not the goal to return only one time per collection?

File: src/api/namespaces/helpers/handlers/collections.ts [hunk 4]
```diff
@@ -1,24 +1,37 @@
...
-        list: {type: 'string', values: ['whitelist', 'blacklist', 'verified', 'nsfw', 'scam']},
+        lists: {type: 'string' },
         sort: {type: 'string', values: ['collection_name', 'list'], default: 'collection_name'},
         order: {type: 'string', values: ['asc', 'desc'], default: 'asc'},
     });
 
-
+    // @TODO: index (collection_name, list)
     const query = new QueryBuilder(`
-                SELECT collection_name, contract, list
```
Comment (blood0cean) [inline src/api/namespaces/helpers/handlers/collections.ts:14]: if the contract will be removed from the return value then also update the example JSON
Screenshot 2022-02-14 at 19 41 39

**Comment by jandrad** [comment]
I think this PR is no longer necessary

**Pull request closed without merging**